internal/provider/resource_gitlab_project_access_token_test.go

```diff
@@ -2,378 +2,103 @@ package provider
+				ImportStateVerifyIgnore: []string{"token"},
 			},
-			//Destroy Project Access Token
+			// Recreate the access token with updated attributes.
 			{
-				Config: testAccGitlabProjectAccessTokenDestroyToken(rInt),
-				Check:  testAccCheckGitlabProjectAccessTokenDoesNotExist(&pat),
-			},
-		},
-	})
-}
-
-func TestAccGitlabProjectAccessToken_accessLevel(t *testing.T) {
-	var pat testAccGitlabProjectAccessTokenWrapper
-	rInt := acctest.RandInt()
-
-	testAccGitlabProjectStart(t)
-
-	resource.Test(t, resource.TestCase{
-		PreCheck:          func() { testAccPreCheck(t) },
-		ProviderFactories: providerFactories,
-		CheckDestroy:      testAccCheckGitlabProjectAccessTokenDestroy,
-		Steps: []resource.TestStep{
-			// Create a project and a Project Access Token
-			{
-				Config: testAccGitlabProjectAccessTokenConfigWithAccessLevel(rInt, accessLevelValueToName[gitlab.MaintainerPermissions]),
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckGitlabProjectAccessTokenExists("gitlab_project_access_token.bar", &pat),
-					testAccCheckGitlabProjectAccessTokenAttributes(&pat, &testAccGitlabProjectAccessTokenExpectedAttributes{
-						name:        "my project token",
-						scopes:      map[string]bool{"read_repository": true, "api": true, "write_repository": true, "read_api": true},
-						expiresAt:   "2022-04-01",
-						accessLevel: accessLevelValueToName[gitlab.MaintainerPermissions],
-					}),
-				),
-			},
-			// Update the Project Access Token to change the parameters
-			{
-				Config: testAccGitlabProjectAccessTokenConfigWithAccessLevel(rInt, accessLevelValueToName[gitlab.DeveloperPermissions]),
+				Config: fmt.Sprintf(`
+				resource "gitlab_project_access_token" "foo" {
+					project = %d
+					name    = "foo"
+					scopes  = ["api", "read_api", "read_repository", "write_repository"]
+					access_level = "developer"
+					expires_at = %q
+				}
+				`, project.ID, time.Now().Add(time.Hour*48).Format("2006-01-02")),
+				// Check computed and default attributes.
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckGitlabProjectAccessTokenExists("gitlab_project_access_token.bar", &pat),
-					testAccCheckGitlabProjectAccessTokenAttributes(&pat, &testAccGitlabProjectAccessTokenExpectedAttributes{
-						name:        "my project token",
-						scopes:      map[string]bool{"read_repository": true, "api": true, "write_repository": true, "read_api": true},
-						expiresAt:   "2022-04-01",
-						accessLevel: accessLevelValueToName[gitlab.DeveloperPermissions],
-					}),
+					resource.TestCheckResourceAttr("gitlab_project_access_token.foo", "active", "true"),
+					resource.TestCheckResourceAttr("gitlab_project_access_token.foo", "revoked", "false"),
+					resource.TestCheckResourceAttrSet("gitlab_project_access_token.foo", "token"),
+					resource.TestCheckResourceAttrSet("gitlab_project_access_token.foo", "created_at"),
+					resource.TestCheckResourceAttrSet("gitlab_project_access_token.foo", "user_id"),
```

Comment:
@armsnyder  - Given we're setting a "expires_at" with a specific value (and this is the only test that does so), it seems like we may want to test that the value matches what we pass in. I know there is an import test right after, but it seems an explicit check here may be good.

Reply:
Same, likely, with the access level (since those explicit tests were removed)

Reply:
After each TestStep, the framework re-runs `terraform plan` and fails the test if the plan is not empty, meaning that the values in state match what you specified in the config. Please correct me if I'm wrong, but I don't think there is a point to using `resource.TestCheckResourceAttr` functions to check what you set in the config.

Reply:
@armsnyder - Ah, I knew it did that in the import steps, I didn't realize it did that automatically after every step, but I suppose that's why the `ExpectNonEmptyPlan` exists logically. 

Seems reasonable, I'm good.

Reply:
Oddly, in the link I shared, their examples do use `resource.TestCheckResourceAttr` to check the attributes set in the config. This is probably because there are edge cases where you might have a `DiffSuppressFunc` that renders the empty-plan check not useful.

My feeling is still that the combination of the test framework's features + using `ImportStateVerify` means we don't need to check these, since `ImportStateVerify` uses strict equality without respecting `DiffSuppressFunc` or `CustomizeDiff`, but I plan on opening an issue so we can discuss more.

Resolving this for now through.